### PR TITLE
SSH API improvements

### PR DESCRIPTION
**Pull request by marineam**
Two features for using SSH:
 - Access to the full ssh.Session API
 - Use the internal SSH agent with external commands

Depends on https://github.com/coreos/mantle/pull/11 and https://github.com/coreos/mantle/pull/10

### CODE REVIEW

File: platform/qemu.go [hunk 1]
```diff
@@ -113,6 +116,21 @@ func (qc *qemuCluster) NewMachine() (Machine, error) {
 		return nil, err
 	}
 
+	// Allow a few authentication failures in case setup is slow.
+	for i := 0; i < 7; i++ {
```
Comment (crawford) [inline platform/qemu.go]: Can you pull this up into a constant?

File: platform/qemu.go [hunk 2]
```diff
@@ -113,6 +116,21 @@ func (qc *qemuCluster) NewMachine() (Machine, error) {
...
+		qm.sshClient, err = qm.qc.SSHAgent.NewClient(qm.IP())
+		if err != nil {
+			fmt.Printf("ssh error: %v\n", err)
+			time.Sleep(time.Second)
```
Comment (crawford) [inline platform/qemu.go]: This too

**Comment by marineam** [comment]
@crawford updated

File: network/ssh.go [hunk 3]
```diff
@@ -59,11 +64,45 @@ func NewSSHAgent(dialer Dialer) (*SSHAgent, error) {
...
+		Dialer:   dialer,
+		User:     defaultUser,
+		Socket:   sockPath,
+		sockDir:  sockDir,
+		listener: listener,
+	}
+
+	go func() {
+		for {
+			conn, err := listener.Accept()
```
Comment (crawford) [inline network/ssh.go]: Does this return an error as soon as `Close()` is called?

Comment (marineam) [inline network/ssh.go]: As far as I know yes, it appeared to be a common go pattern but I don't personally know the details (yet)

**Comment by crawford** [comment]
LGTM

**Comment by pbx0** [comment]
LGTM

**Pull request merged**